codegen/templates/generated.gotpl

```diff
@@ -48,12 +64,42 @@ type DirectiveRoot struct {
 type executableSchema struct {
 	resolvers  ResolverRoot
 	directives DirectiveRoot
+	complexity ComplexityRoot
 }
 
 func (e *executableSchema) Schema() *ast.Schema {
 	return parsedSchema
 }
 
+func (e *executableSchema) Complexity(typeName, field string, childComplexity int, rawArgs map[string]interface{}) (int, bool) {
+	switch typeName + "." + field {
+	{{ range $object := .Objects }}
+		{{ range $field := $object.Fields }}
+			{{ if $field.CustomComplexity }}
+				case "{{$object.GQLType}}.{{$field.GQLName}}":
+				    if e.complexity.{{$object.GoType}}.{{$field.GoFieldName}} == nil {
+						break
+					}
+					{{ if . }}args := map[string]interface{}{} {{end}}
```

Comment:
This is basically copied from `args.gotpl`. There's probably a better way to do this with less duplication.

In general, it seems like it might be too expensive to be doing all the argument conversion and unmarshaling twice, as well. (Now it will happen once for the complexity calculation, and then again for the actual query execution.)